.github/workflows/azure-e2e-test.yml

```
@@ -116,42 +119,76 @@ jobs:
           MANAGED_IDENTITY_NAME="${{ secrets.AZURE_MANAGED_IDENTITY_NAME}}"
           CAA_IMAGE="${CAA_IMAGE}"
         EOF
-        cat /tmp/provision_azure.properties
+        cat ${{ env.TEST_PROVISION_FILE }}
 
     - name: Create public ssh key and secrets
       run: |
         # TODO: This is a temporary fix to create an empty file for env.
         touch install/overlays/azure/service-principal.env
         ssh-keygen -t rsa -b 4096 -f install/overlays/azure/id_rsa -N "" -C dev@coco.io
 
+    - name: Save the configuration created here
+      uses: actions/cache/save@v3
+      with:
+        path: |
+          install/overlays/azure
+          ${{ env.TEST_PROVISION_FILE }}
+        key: caa-config-${{ github.run_id }}-${{ github.run_attempt }}
+
```

It's probably more idiomatic to use [artifacts][(https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#passing-data-between-jobs-in-a-workflow)) for this:

```suggestion
    - name: Save the configuration created here
      uses: actions/upload-artifact@v3
      with:
        path: |
          install/overlays/azure
          ${{ env.TEST_PROVISION_FILE }}
        name: e2e-configuration
```